### PR TITLE
[Contrib][Cblas] Avoid allocating memory on every call to cblas batch gemm

### PR DESCRIPTION
**Pull request by soiferj**
This change prevents memory allocation on each call to CblasSgemmBatchOp and CblasDgemmBatchOp.

@ajtulloch @yinghai would you be able to take a look?

### CODE REVIEW

**Comment by icemelon9** [comment]
Could you share some measurement on how much this change affects the latency?

File: src/contrib/cblas/cblas.cc [hunk 1]
```diff
@@ -67,9 +74,12 @@ struct CblasSgemmBatchOp {
     CBLAS_TRANSPOSE trans_a = BooleanToTranspose(ta);
     CBLAS_TRANSPOSE trans_b = BooleanToTranspose(tb);
 #if USE_MKL_BLAS == 1
-    std::vector<const float*> A_array(batch_size);
-    std::vector<const float*> B_array(batch_size);
-    std::vector<float*> C_array(batch_size);
+    static std::vector<const float*> A_array;
```
Comment (yinghai) [inline src/contrib/cblas/cblas.cc]: `static` is not thread safe here.

Comment (soiferj) [reply src/contrib/cblas/cblas.cc]: AFAIK, TVM runs all ops sequentially, so two calls to BatchMatMul cannot happen in parallel. What do you think?

**Comment by soiferj** [comment]
Weirdly, this doesn't seem to have any noticeable perf impact in my tests :( I'll close it for now. Thanks everyone.

File: src/contrib/cblas/cblas.cc (hunk 1 above)
Comment (yinghai) [reply src/contrib/cblas/cblas.cc]: There is usecase that upper layer service launches multiple TVM runtime, each itself is single threaded on one thread.

**Pull request closed without merging**